pkg/restapi/operation/common/msgservice.go

```diff
@@ -73,15 +80,31 @@ func (m *msgService) Accept(msgType string, purpose []string) bool {
 }
 
 func (m *msgService) HandleInbound(msg service.DIDCommMsg, myDID, theirDID string) (string, error) {
-	if m.name == "" {
+	if m.name == "" || m.topicHandle == nil {
 		return "", fmt.Errorf(errTopicNotFound)
 	}
 
-	// TODO: get rid of this  msg.(service.DIDCommMsgMap)
-	bytes, err := json.Marshal(&inboundMsg{Message: msg.(service.DIDCommMsgMap), MyDID: myDID, TheirDID: theirDID})
+	bytes, err := m.topicHandle(msg, myDID, theirDID)
 	if err != nil {
 		return "", fmt.Errorf(errMsgSvcHandleFailed, err)
 	}
 
 	return "", m.notifier.Notify(m.name, bytes)
 }
+
+// genericHandleFunc handle function for converting incoming messages to generic topic.
+func genericHandleFunc() handleFunc {
+	return func(msg service.DIDCommMsg, myDID, theirDID string) ([]byte, error) {
+		topic := struct {
+			Message  service.DIDCommMsgMap `json:"message"`
+			MyDID    string                `json:"mydid"`
+			TheirDID string                `json:"theirdid"`
+		}{
+			msg.(service.DIDCommMsgMap),
```

Comment:
TODO?

Reply:
You are right, had a TODO in previous version.
Passed incoming `DIDCommMsg + DIDs` as it is to webhook as bytes, and webhook will take care of marshalling it by topic.